core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java

```diff
@@ -192,6 +194,11 @@ public class Dispatcher {
      * Store ConfigurationManager instance, set on init.
      */
     protected ConfigurationManager configurationManager;
+    private ObjectFactory objectFactory;
```

Comment:
Maybe they should be moved into constructor?

Reply:
Since the Dispatcher itself is not container-managed, we can't use constructor injection for it

Reply:
It doesn't have to be, just using `container.inject(Dispatcher.class)` will create a new instance and injects all the dependencies. Anyway this would be a bit out of scope for this PR.

Reply:
Yep would require a major refactor, I've got some ideas but not a high priority right now, as long as everything works